test/unit/math/prim/prob/wiener_full_test.cpp

```diff
@@ -0,0 +1,275 @@
+#include <stan/math/prim/functor.hpp>
```

Comment:
You also need `mix` tests for your functions. See the [`bessel_first_kind` test](https://github.com/stan-dev/math/blob/98117fa4616d50d9544fed882c5a5c2b9803cf85/test/unit/math/mix/fun/bessel_first_kind_test.cpp#L6) for an example of how to construct the test to only assess a subset of parameters at a time (as the autodiff testing framework can only test three parameters at a time)

Reply:
One question here:
We had a chat with @bob-carpenter on the topic testing [here](https://discourse.mc-stan.org/t/testing-for-7-parameter-function-with-known-partials/27046/11). He said we would only need unit tests, therefore we implemented the ones existing. In our functions, we only accept scalar-input. Shall we still add `mix` tests? Could you shortly explain what the content of a `mix` test should be?

Reply:
Yep, the `mix` tests are part of the unit tests - they test that your functions work with various combinations of `double`, `var`, and `fvar<T>` types (i.e., higher-order autodiff). For an example, see the [mix test for normal_lpdf](https://github.com/stan-dev/math/blob/develop/test/unit/math/mix/prob/normal_test.cpp)

Reply:
Thanks a lot for your suggestions. We ran into problems with `expect_ad` as we can only provide first derivatives, so higher-order derivatives are not supported by this function. As far as I can tell, using autodiff to provide them is not possible, as the numerical integration and the autodiff don't work well together (at least they didn't for the first derivatives, which is why we provide them manually). This leads to two questions:
a) How should we handle the case where `fvar<T>` is passed to the function (right now, the compiler throws an error)?
b) How should we structure the mix tests? One idea would be to move the manual tests for the first derivatives to mix and not use the autodiff testing framework. Do you have other suggestions?